### PR TITLE
PYTHON-4480 Deprecate create=True for Collection

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

File: test/test_collection.py [hunk 1]
```diff
@@ -197,12 +197,6 @@ def test_create(self):
             lambda: "create_test_no_wc" in db.list_collection_names(),
             "create create_test_no_wc collection",
         )
-        db.create_test_no_wc.drop()
```
Comment (blink1073) [inline test/test_collection.py:200]: We should keep this and catch the warning.

File: pymongo/asynchronous/collection.py [hunk 2]
```diff
@@ -248,6 +249,11 @@ def __init__(
 
         if create or kwargs:
             if _IS_SYNC:
+                warnings.warn(
+                    "The `create` and `kwargs` arguments are deprecated and will be removed in PyMongo 5.0",
```
Comment (ShaneHarvey) [inline pymongo/asynchronous/collection.py:253]: I would add "arguments to Collection are..."

File: test/test_collection.py (hunk 1 above)
Comment (ShaneHarvey) [reply test/test_collection.py:200]: Correct, when we deprecate something we keep the tests and also add an extra assertion to ensure the warning is actually raised.

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**